### DISCUSSION TITLE
custom highlight groups - hl_add looks like its not adding new groups

### DISCUSSION THREAD

**Discussion by vic-gfuentes** [NvChad/NvChad#2974] [Q&A] [answered]
Hi! Quick question, I have the following config:
```lua
-- chadrc.lua
M.base46 = {
  theme = "onedark",
  hl_override = highlights.override,
  hl_add = highlights.add,
}
```
```lua
-- highlights.lua
local M = {}

M.override = {
  Comment = {
    italic = true
  },
}

M.add = {
  NvimTreeOpenedFolderName = { fg = "green", bold = true },
  ConflictIncoming = { bg = "#203242" },
  ConflictCurrent = { bg = "#1f3834" },
}

return M
```

Im trying to use those groups in a plugin `akinso/git-conflict.nvim`
```lua
  {
    "akinsho/git-conflict.nvim",
    version = "*",
    event = "BufRead",
    config = function()
      require("git-conflict").setup {
        default_mappings = true,
        disable_diagnostics = false,
        highlights = {
          incoming = "ConflictIncoming",
          current = "ConflictCurrent",
        },
      }
    end,
  },
```

The problem here is that when I load `nvim` and check the groups I see that my custom groups were not loaded I assume this has something to do with Lazy loading but im not sure. As soon as I open a file throws an error since as you may imagine the group does not exist.

I wonder if this can be solved using integrations: [https://nvchad.com/docs/config/theming#integrations](https://nvchad.com/docs/config/theming#integrations), but being real honest I have no idea how to implement this

**Comment by siduck** [comment]
you need @comment tooo.. only Comment wont work

and dont add already existing hlgroups in hl-add
![image](https://github.com/user-attachments/assets/ae75cd3e-4a4e-4cb5-a0b9-e08d2a4af322)

**Reply by vic-gfuentes** [reply]
My bad on the existing one, got rid of that one since I will no longer use it, also got rid of the highlights file and now I only have this in my chadrc:

```lua
local M = {}

M.base46 = {
  theme = "onedark",
  hl_add = {
    ConflictIncoming = { bg = "#203242" },
    ConflictCurrent = { bg = "#1f3834" },
  }
}

M.ui = {
  statusline = {
    theme = "minimal",
    separator_style = "default"
  }
}

return M

```

Sad thing is, still not working. 

@siduck what do you mean by I need @comment too? lost the track a little bit. Also thanks for helping, really appreciate it

**Reply by vic-gfuentes** [reply]
Update -

Cleaned these folders:
```bash
rm -rf ~/.local/state/nvim
rm -rf ~/.local/share/nvim
```

Did a fresh install basically and that solved the problem, something wicked might have happened. Thanks @siduck

**Reply by siduck** [reply, marked as answer]
@vic-gfuentes  
![image](https://github.com/user-attachments/assets/da2aa239-7527-4a24-9ea0-44c679d36cbd)
